airflow/contrib/hooks/ftp_hook.py

```diff
@@ -226,7 +226,11 @@ def rename(self, from_name, to_name):
     def get_mod_time(self, path):
         conn = self.get_conn()
         ftp_mdtm = conn.sendcmd('MDTM ' + path)
-        return datetime.datetime.strptime(ftp_mdtm[4:], '%Y%m%d%H%M%S')
+        time_val = ftp_mdtm[4:]
+        #time_val optionally has microseconds
+        if "." in time_val:
```

Comment:
I'm not sure, according to the RFC:
```
The "." and subsequent digits ("sss") are optional.  However the "."
MUST NOT appear unless at least one following digit also appears.
```
So testing if a `.` is present seems to be the best option here. If a `.` is present it MUST be parsable with `%Y%m%d%H%M%S.%f`. If it isn't parsable, then it also won't be parsable with `%Y%m%d%H%M%S` because there is a `.` in the `time_val` string